mempool/reactor.go

```diff
@@ -28,6 +29,17 @@ const (
 	maxActiveIDs = math.MaxUint16
 )
 
+var (
+	// proto overhead for the message with no transactions
+	// see txs
+	emptyMessage = protomem.Message{
+		Sum: &protomem.Message_Txs{
+			Txs: &protomem.Txs{},
+		},
+	}
+	emptyMessageSize = emptyMessage.Size() + 2 // 2 -> Txs key (0 above)
```

Comment:
@marbar3778 does this make sense to you?

Reply:
From skimming the [documentation](https://developers.google.com/protocol-buffers/docs/encoding#packed), I think the overhead of the `Txs` field will depend on the total size of the transactions in it (the payload size) in addition to the length-prefixing.

Trying to predict the encoded size seems a bit brittle to me, would it be possible to simply build the message as we go along and check the actual message size as reported by Protobuf?

Reply:
> I think the overhead of the Txs field will depend on the total size of the transactions

only primitive types can be "packed"

Reply:
Ah, you're right. I guess we'll have to add on the size of the field key for each tx then?

Reply:
> Trying to predict the encoded size seems a bit brittle to me, would it be possible to simply build the message as we go along and check the actual message size as reported by Protobuf?

we do it in a few places unfortunately. checking actual message size is an option, but I decided against it because marshalling on every tx will be expensive

Reply:
> marshalling on every tx will be expensive

We don't have to marshal, just call `Size()` on the message struct.

Reply:
Isn't what the Size does? marshals the message and calls `len()`?

Reply:
No, it calculates the size. Here's the code for `Txs`:

```go
func (m *Txs) Size() (n int) {
	if m == nil {
		return 0
	}
	var l int
	_ = l
	if len(m.Txs) > 0 {
		for _, b := range m.Txs {
			l = len(b)
			n += 1 + l + sovTypes(uint64(l))
		}
	}
	return n
}
```

Which is basically what we're trying to emulate.